### PR TITLE
fix: add cache to generateGFunction

### PR DESCRIPTION
**Pull request by nodece**
Signed-off-by: Zixuan Liu <nodeces@gmail.com>

Fix: https://github.com/casbin/node-casbin/issues/189

### CODE REVIEW

**Review by GopherJ — APPROVED** [review]

File: src/util/builtinOperators.ts [hunk 1]
```diff
@@ -233,19 +233,29 @@ function globMatch(string: string, pattern: string): boolean {
 
 // generateGFunction is the factory method of the g(_, _) function.
 function generateGFunction(rm: rbac.RoleManager): any {
+  const memorized = new Map<string, boolean>();
```
Comment (hsluoyz) [inline src/util/builtinOperators.ts:236]: Will `memorized` be still available to access when `generateGFunction()` function exists? If yes, then merge.

**Review by hsluoyz — COMMENTED** [review]
Will `memorized` be still available to access when `generateGFunction()` function exists? If yes, then merge.

File: src/util/builtinOperators.ts (hunk 1 above)
Comment (nodece) [reply src/util/builtinOperators.ts:236]: @hsluoyz `memorized` used to cache the results of `g()`, each call to `generateGFunction()` generates a new cache in enforcer.

**Comment by hsluoyz** [comment]
lgtm.

**Pull request merged**